### PR TITLE
Concurrent renderings

### PR DESCRIPTION
**Pull request by aab-odoo**

### CODE REVIEW

File: src/misc/async_root.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Component } from "../component/component";
+import { xml } from "../tags";
```
Comment (aab-odoo) [inline src/misc/async_root.ts]: no doc?

File: src/component/component.ts [hunk 2]
```diff
@@ -664,47 +649,15 @@ export class Component<T extends Env, Props extends {}> {
...
-  }
 }
 
 //------------------------------------------------------------------------------
 // Error handling
 //------------------------------------------------------------------------------
 
+Fiber.prototype.handleError = function(error) {
+  errorHandler(error, this);
+};
```
[the remark this answers is not in the document]
Comment (ged-odoo) [reply src/component/component.ts]: cyclic dependencies... if you do that, rollup will throw a warning

File: src/component/fiber.ts [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    component.__owl__.currentFiber = this;
+  }
+
+  /**
+   * In some cases, a rendering initiated at some component can detect that it
+   * should be part of a larger rendering initiated somewhere up the component
+   * tree.  In that case, it needs to cancel all the previous rendering and
+   * remap itself as a part of the previous rendering.
+   */
+  __remapFiber(oldFiber: Fiber) {
```
[the remark this answers is not in the document]
Comment (ged-odoo) [reply src/component/fiber.ts]: yeah, __ is only for component I think

File: src/misc/async_root.ts (hunk 1 above)
Comment (ged-odoo) [reply src/misc/async_root.ts]: oups

File: src/component/component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (aab-odoo) [reply src/component/component.ts]: so should we not exporting it?

[the remark this answers is not in the document]
Comment (ged-odoo) [reply src/component/component.ts]: yes, just did it

File: src/component/directive.ts [hunk 4]
```diff
@@ -186,7 +186,7 @@ QWeb.utils.defineProxy = function defineProxy(target, source) {
 
 QWeb.addDirective({
   name: "component",
-  extraNames: ["props", "keepalive", "asyncroot"],
```
[the remark this answers is not in the document]
Comment (ged-odoo) [reply src/component/directive.ts]: this is done with the t-asyncroot change. but this should have been a separate commit

**Pull request merged**